api/logger.js

```diff
@@ -0,0 +1,58 @@
+const winston = require('winston');
+
+const loggers = {};
+
+module.exports = name => {
+  if (!loggers[name]) {
+    const transports = {};
+
+    if (process.env.LOG_FILE === 'true') {
+      transports.file = {
+        filename: 'hitech_api.log',
+        label: name,
+        level: process.env.LOG_LEVEL,
+        maxFiles: 5,
+        maxsize: 2097152,
+        tailable: true,
+        timestamp: true,
+        zippedArchive: true
+      };
+    }
+
+    if (process.env.LOG_CONSOLE === 'true') {
+      transports.console = {
+        colorize: true,
+        label: name,
+        level: process.env.LOG_LEVEL,
+        prettyPrint: true,
+        silent: process.argv.includes('--silent'),
+        timestamp: true
+      };
+    } else {
+      transports.console = {
+        silent: true
+      };
+    }
+
+    winston.loggers.add(name, transports);
+    loggers[name] = true;
+  }
+
+  const logger = winston.loggers.get(name);
+  // Override the log function to append request information
+  // if it's provided
+  logger.log = (level, req, ...args) => {
+    let requestInfo = req;
+    if (req && typeof req !== 'string') {
+      requestInfo = {
+        requestID: req.id
+      };
+      if (req.user) {
+        requestInfo.userID = req.user.id;
+      }
+    }
+    winston.Logger.prototype.log.apply(logger, [level, ...args, requestInfo]);
```

Comment:
I opted to reverse the log message and the request info because it looks cleaner in the output.  The result is a log line that looks about like this:

```
2018-02-23T16:59:49.680Z - silly: [auth middleware] got a can middleware request for [create-roles]
{ requestID: 'f57e02e0-18ba-11e8-9ad0-893dd6149dba' }
```

Reply:
nice!

Reply:
#hefancy